services/web/server/src/simcore_service_webserver/projects/project_lock.py

```diff
@@ -0,0 +1,50 @@
+from typing import Dict, Optional, Union
+
+import aioredlock
+from aiohttp import web
+from models_library.projects import ProjectID
+from models_library.projects_state import Owner, ProjectLocked, ProjectStatus
+from simcore_service_webserver.resource_manager.redis import (
+    get_redis_lock_manager,
+    get_redis_lock_manager_client,
+)
+
+PROJECT_REDIS_LOCK_KEY: str = "project:{}"
+
+
+async def lock_project(
+    app: web.Application,
+    project_uuid: Union[str, ProjectID],
+    status: ProjectStatus,
+    user_id: int,
+    user_name: Dict[str, str],
+) -> aioredlock.Lock:
```

Comment:
VERY nice! 
It is not obvious to me that this return a context managed object, i would add a little doc describing usage. Something like: create a context to keep project locked
```python
async with await lock_project(...):
    # do some stuff with project, nobody can change it
```